src/main/steps/applicant1/hub-page/post.ts

```diff
@@ -1,11 +1,15 @@
 import autobind from 'autobind-decorator';
 
-import { APPLICANT_1_CONFIRM_RECEIPT } from '../../../app/case/definition';
+import { APPLICANT_1_CONFIRM_RECEIPT, ApplicationType, DRAFT_CONDITIONAL_ORDER } from '../../../app/case/definition';
+import { AppRequest } from '../../../app/controller/AppRequest';
 import { AnyObject, PostController } from '../../../app/controller/PostController';
 
 @autobind
 export default class HubPagePostController extends PostController<AnyObject> {
-  protected getEventName(): string {
-    return APPLICANT_1_CONFIRM_RECEIPT;
+  protected getEventName(req: AppRequest<AnyObject>): string {
+    return req.session.userCase.applicationType === ApplicationType.JOINT_APPLICATION &&
+      req.body.applicant1ConfirmReceipt
+      ? APPLICANT_1_CONFIRM_RECEIPT
```

Comment:
this doesn't look correct to me... but I might be wrong.

If you submit the from the hub page surely there will be different events based on the state of the case? Drafting the CO, drafting the FO and also confirming receipt. Doing it based on whether it's a joint app doesn't seem right.

Reply:
currently theres only 2 events and the logic was based on if its sole or joint. ofc down the line if theres more then itll change but for now its based on that, i can change it to be based on states and application types.